### ISSUE TITLE
tm: bad error in log

### ISSUE THREAD

**Issue by lazedo** [kamailio/kamailio#1727]
### Description
tm module writes a misleading log error that `triggers` ops alerts.

#### Reproduction
using this partial script

```
event_route[core:worker-one-init]
{
    async_route("DEFERRED_CLEANUP", "20");
}

route[DEFERRED_CLEANUP]
{
  xlog("L_INFO", "processing deferred cleanup\n");
}
```

#### Log Messages

```
2018-11-15T19:22:44.547049+00:00 apps001 kamailio[2105]: INFO: <script>: processing deferred cleanup
2018-11-15T19:22:44.554085+00:00 apps001 kamailio[2105]: ERROR: tm [t_reply.c:551]: _reply_light(): no resolved dst to send reply to
```

### Possible Solutions

i silence the error with the below patch but i'm pretty sure its not the proper way to do it.

``` 
diff --git a/src/modules/tm/t_reply.c b/src/modules/tm/t_reply.c
index 8a95fe7..99c59db 100644
--- a/src/modules/tm/t_reply.c
+++ b/src/modules/tm/t_reply.c
@@ -548,7 +548,9 @@
 	 * the chances for this increase a lot.
 	 */
 	if (unlikely(!trans->uas.response.dst.send_sock)) {
-		LM_ERR("no resolved dst to send reply to\n");
+		if(is_local(trans)) {
+			LM_ERR("no resolved dst to send reply to\n");
+		}
 	} else {
 		if (likely(SEND_PR_BUFFER( rb, buf, len )>=0)){
 			if (unlikely(code>=200 && !is_local(trans) &&
```

### Additional Information

  * **Kamailio Version** - output of `kamailio -v`

```
version: kamailio 5.3.0-dev0 (x86_64/linux) 37bff4-dirty
flags: STATS: Off, USE_TCP, USE_TLS, USE_SCTP, TLS_HOOKS, USE_RAW_SOCKS, DISABLE_NAGLE, USE_MCAST, DNS_IP_HACK, SHM_MEM, SHM_MMAP, PKG_MALLOC, Q_MALLOC, F_MALLOC, TLSF_MALLOC, DBG_SR_MEMORY, USE_FUTEX, FAST_LOCK-ADAPTIVE_WAIT, USE_DNS_CACHE, USE_DNS_FAILOVER, USE_NAPTR, USE_DST_BLACKLIST, HAVE_RESOLV_RES
ADAPTIVE_WAIT_LOOPS=1024, MAX_RECV_BUFFER_SIZE 262144 MAX_URI_SIZE 1024, BUF_SIZE 65535, DEFAULT PKG_SIZE 8MB
poll method support: poll, epoll_lt, epoll_et, sigio_rt, select.

```

* **Operating System**:

```
Linux 2.6.32-754.el6.x86_64 #1 SMP Tue Jun 19 21:26:04 UTC 2018 x86_64 x86_64 x86_64 GNU/Linux
```

**Comment by miconda** [comment]
Async was not designed to work with a faked request, like the one from `event_route[core:worker-one-init]`, because it does suspend/continue of a transaction for a request that is not received from the network and not to be routed. Maybe it can be enhanced, but meanwhile look at the next alternatives to execute actions after a period of time:

  * rtimer module
  * evrexec module

**Comment by lazedo** [comment]
@miconda i looked at rtimer and evrexec in the first place. the problem is that we need a one time deferred execution

**Comment by miconda** [comment]
evrexec executes only once, you would have to do a loop or something else to continue execution. It allows to set the interval to wait before executing the route block.

With rtimer, you can use a test on a $var(...), which is initialized to 0, the set it to 1 with the first execution. At the top of the route you should do if $var(...) != 0 then return.

I also pushed a patch to async module to detect when faked msg is tried to be used and return error. I will look at other parts of code that are similar and try to enforce it there as well.

I am going to close this one here, the error is not in tm anyhow. We can discuss more on mailing list if you have a specific use case and you don't know if it is possible to do/how to do.

**Comment by lazedo** [comment]
@miconda i'm ok with you closing this one and discussion going into mailing list.

fyi, none of the proposed uses (rtimer, evrexec) will keep you from having a `ghost` process that does nothing after 1st execution, which seems not ideal since it still using resources (memory).
rtimer will keep running the event routes
evrexec will never exit the process

**Comment by miconda** [comment]
rtimer can use the core timer, no new process created.

Also async creates processes, so if you just use it for this case, then there are ghost processes in the same way.

See also timer module, wrote by ser developers, iirc, it has a way to disable the timer task, but I am not familiar with it to say for sure.

Then, feel free to add a new feature for what you need. Just that what you reported here is not a bug and not the right fix.

**Comment by lazedo** [comment]
yes, already agreed on closing this.
thanks
